common/azure-common/src/main/java/com/azure/common/implementation/util/ImplUtils.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.azure.common.implementation.util;
+
+/**
+ *  The util class is a helper class for clone operation.
+ */
+public class ImplUtils {
+
+    private ImplUtils() {
+        // Exists only to defeat instantiation.
+    }
+
+    /**
+     * Creates a copy of the source byte array.
+     * @param source Array to make copy of
+     * @return A copy of the array, or zero length array if source was null.
+     */
+    public static byte[] clone(byte[] source) {
+        if (source == null) {
+            return new byte[0];
```

Comment:
I've the same concerns here as I mentioned previously. Bytes array that is zero length (EMPTY) is different from No bytes at all (NULL). EMPTY means array exists, but it contains no value and NULL means the array does not exist. IMO the core utils like shouldn't break this assumption.

Reply:
For most of our major cases, the zero length spotbugs check turns to false positive. Can we suppress this from spotbugs? @JonathanGiles 
Otherwise, we will have too many filter rules on this and spend too much time in addressing the bugs back and forth.